### PR TITLE
Socket test improvements

### PR DESCRIPTION
**Pull request by rnro**
This PR aims to make Socket tests faster and more stable, no source changes.

- Servers now bind port 0 and report `listeningPort`.
- Where the API doesn't expose the port a test helper is used to discover a free port and that is then used (`discoverFreeLoopbackPorts`). This is still racy, but avoids most collisions and issues when hanging tests have the port.
- `TCPClientHarness.init` reports a start failure instead of dropping it with `try?`, so a bind failure no longer surfaces later as an unexplained timeout.
- The refused-connection test asserts from its recorded state history rather than an inverted expectation, which XCTest can only satisfy by waiting out the whole timeout (locally 0.002s against 5.003s).
- The split-send server blocks on the peer's FIN instead of a one second sleep.
- The interface test which previously exhausted all FDs on the system now lowers its own `RLIMIT_NOFILE` instead of opening sockets until the kernel refuses. This was probably a cause of a reasonable amount of test flakiness and was slow. (locally 17s to under a millisecond).
- Three waits are deliberately kept, each now commented with rationale

### CODE REVIEW

File: Tests/SwiftNetworkTests/SwiftNetworkSocketTests.swift [hunk 1]
```diff
@@ -221,21 +223,21 @@ final class SwiftNetworkSocketTests: NetTestCase {
     // MARK: - Edge case payloads (UDP)
 
     func testEmptyPayload() {
-        let harness = UDPLoopbackHarness(basePort: 10950)
+        let harness = UDPLoopbackHarness()
```
Comment (agnosticdev) [inline Tests/SwiftNetworkTests/SwiftNetworkSocketTests.swift:226]: This is nice, thank you!

**Review by agnosticdev — APPROVED** [review]

File: Tests/SwiftNetworkTests/SwiftNetworkInterfaceTests.swift [hunk 2]
```diff
@@ -17,12 +17,9 @@ import XCTest
 
 #if canImport(Glibc)
 import Glibc
-internal import Logging
 #elseif canImport(Musl)
 import Musl
-internal import Logging
 #elseif canImport(os)
-internal import os
```
Comment (PayasR) [inline Tests/SwiftNetworkTests/SwiftNetworkInterfaceTests.swift:25]: Is this just an empty #elseif branch now?

**Review by PayasR — APPROVED** [review]

File: Tests/SwiftNetworkTests/SwiftNetworkInterfaceTests.swift (hunk 2 above)
Comment (rnro) [reply Tests/SwiftNetworkTests/SwiftNetworkInterfaceTests.swift:25]: Good spot

**Review by tfpauly — APPROVED** [review]
